### DISCUSSION TITLE
Build fails after removing an app (unable to parse workspace package.json)

### DISCUSSION THREAD

**Discussion by milovangudelj** [vercel/turborepo#7191] [Help]
I had three apps in my repository. I'll call them `A`, `B` and `B-v2`.

I made a major refactor in  a different repo with the same structure and to ease in the merge of the changes I copied `B` from the refactor repo and named it `B-v2` in my repo. Once I finished moving everything and checked that it was working as expected I removed `B-v2` from my repository.

Now if I run `pnpm build` I get this error:
```terminal
 ERROR  run failed: unable to parse workspace package.json: unable to read package.json: failed to open file `locationofmyrepo/apps/B-v2/package.json`
 ELIFECYCLE  Command failed with exit code 1.
```

I tried removing the repo and cloning it again, but it did nothing. I removed all build and node artifacts, and that too did nothing. I even tried logging out of turbo from the cli cause maybe it had something to do with remote caching, but nada.

`B-v2` doesn't exist anymore, so why does turbo still have a reference to it somewhere? I even tried looking for the app name in the repo and it's nowhere to be found.

How can I clean Turbo's cache system wide? Maybe there's a reference to that directory somewhere in Turbo's files. Or maybe it has to do with pnpm. I really don't know...

**Comment by milovangudelj** [comment]
Adding back a dummy `package.json` where Turbo expects to find it makes running scripts possible again, but it's a temporary workaround.

**Reply by milovangudelj** [reply]
Update: restarting my pc solved the issue. I don't know why tho... :sweat_smile:

**Comment by OtisTemler** [comment]
Hi! As I happen to have the very same issue: It doesn't seem to be the pnpm cache but instead turbo's.

So all I did was to edit the dev command in the package.json to add `--force` at the end. So  `"dev": "turbo dev --force"`, after this start the dev server once `pnpm run dev` and it should just start just fine. After you started the dev servers once, you can remove the `--force` again.

**Reply by milovangudelj** [reply]
Thanks for the suggestion, but I don't think it's an issue with Turbo's cache for a few reasons.

First, the dev pipeline is set up in a way that it should already ignore the cache when running (`"cache": false`). Then running a build with the `--force` flag still led to the same error. And removing the entire directory and cloning the repo again did not solve the problem.

It was resolved only once I restarted my pc.

Maybe it has something to do with the way pnpm resolves your workspaces from `pnpm-workspace.yaml`. I have it set up with a generic `'apps/*'` entry and it could have potentially cached the result of that expansion somewhere.

**Reply by OtisTemler** [reply]
True that. Hmmm, for me it worked tho - but maybe changing the package.json command alone is enough then for pnpm to throw away it's cache? - Dunno.

`pnpm store prune` at least didn't help, that I remembered. All a big mystery.

**Reply by milovangudelj** [reply]
Well, I think we'll never know what's the root cause of this issue... 😅
